name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/util/StringToCurrencyConverter.java

```diff
@@ -84,20 +90,47 @@ private long convertToLong(String part) throws ParseException
 
         String strAfter = m.group(3);
         long after = 0;
-        if (strAfter != null && strAfter.length() > 0)
+        if(strAfter == null) 
         {
-            int length = (int) Math.log10(factor);
-
-            if (strAfter.length() > length)
-                strAfter = strAfter.substring(0, length);
-
-            after = Long.parseLong(strAfter);
-
-            for (int ii = strAfter.length(); ii < length; ii++)
-                after *= 10;
+            // the input has no decimal part
+            if ("BE".equals(Locale.getDefault().getCountry())) //$NON-NLS-1$
+            {
+                // In some culture (eg. fr_be and nl_be) it is normal do use the group separator as the decimal separator in the input field
+                // Other applications like Excel convert this automatically to the correct value
+                // So we check if there is only one grouping separator in the string
+                
+                String[] groups = strBefore.split(Pattern.quote(Character.toString(this.groupingSeperator)));
+                if(groups.length == 2)
+                {
+                    // We found only one grouping separator so we assume this is the decimal separator
+                    before = full.parse(groups[0]);
```

Comment:
If the dot is the first character in the string (Example ".23"), then this parsing will fail.
You will have to manually check if ```groups[0].length() > 0```